### ISSUE TITLE
TODO: binary crossentropy stability in JAX and torch

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras-core#196]
We implement BCE as follows:

```
def binary_crossentropy(target, output, from_logits=False):
    target = jnp.array(target)
    output = jnp.array(output)

    if target.shape != output.shape:
        raise ValueError(
            "Arguments `target` and `output` must have the same shape. "
            "Received: "
            f"target.shape={target.shape}, output.shape={output.shape}"
        )

    if from_logits:
        output = jnn.sigmoid(output)

    output = jnp.clip(output, epsilon(), 1.0 - epsilon())
    bce = target * jnp.log(output)
    bce += (1.0 - target) * jnp.log(1.0 - output)
    return -bce
```

An issue here is the fact that, provided logits, we take the sigmoid, and then the log. This can lead to numerical instability in fp16.

Can we come up with an implementation that goes through `log_sigmoid` or something?
